src/autorange.jl

```diff
@@ -0,0 +1,54 @@
+function autorange(img, tform)
+    R = CartesianRange(indices(img))
+    I = first(R)
+    mn = mx = x = tform(SVector(I))
+    for I in CornerIterator(R)
+        x = tform(SVector(I))
+        # we map min and max to prevent type-inference issues
+        # (because min(::SVector,::SVector) -> Vector)
+        mn = map(min, x, mn)
+        mx = map(max, x, mx)
+    end
+    _autorange(convert(Tuple, mn), convert(Tuple, mx))
+end
+
+@noinline _autorange(mn,mx) = map((a,b)->floor(Int,a):ceil(Int,b), mn, mx)
+
+## Iterate over the corner-indices of a rectangular region
+immutable CornerIterator{I<:CartesianIndex}
+    start::I
+    stop::I
+end
+CornerIterator{I<:CartesianIndex}(R::CartesianRange{I}) = CornerIterator{I}(R.start, R.stop)
+
+eltype{I}(::Type{CornerIterator{I}}) = I
+iteratorsize{I}(::Type{CornerIterator{I}}) = Base.HasShape()
+
+size{N}(iter::CornerIterator{CartesianIndex{N}}) = ntuple(d->iter.stop.I[d]-iter.start.I[d]==0 ? 1 : 2, Val{N})
```

Comment:
I added this function to complete the required iterator interface. I have yet to write formal test, but this code seems to do it and is type stable